src/include/column.i

```diff
@@ -122,13 +122,18 @@ __col_insert_search_match(WT_INSERT_HEAD *ins_head, uint64_t recno)
 			continue;
 		}
 
-		ins_recno = WT_INSERT_RECNO(*insp);
+		/*
+		 * Use a local variable to access the insert because the skip
+		 * list can change across references.
+		 */
+		ret_ins = *insp;
+		ins_recno = WT_INSERT_RECNO(ret_ins);
 		cmp = (recno == ins_recno) ? 0 : (recno < ins_recno) ? -1 : 1;
 
 		if (cmp == 0)			/* Exact match: return */
-			return (*insp);
+			return (ret_ins);
 		if (cmp > 0)			/* Keep going at this level */
-			insp = &(*insp)->next[i];
+			insp = &ret_ins->next[i];
```

Comment:
On the surface this looks correct to me. Two comments:

1. We're still reading `*insp` twice and presumably it can change between the two reads, so I'd suggest:
```
/*
 * Use a local variable to access the insert because the skip
 * list can change across references.
 */
if ((ret_ins = *insp) == NULL) {
        --i;
        --insp;
        continue;
}

ins_recno = WT_INSERT_RECNO(ret_ins);
cmp = (recno == ins_recno) ? 0 : (recno < ins_recno) ? -1 : 1;

if (cmp == 0)                   /* Exact match: return */
```

2. The two functions in the same file above this one, `__col_insert_search_lt()` and `__col_insert_search_gt()` were probably copied from `__col_insert_search_match()`, and appear to need the same fix? The function `__col_insert_search()` seems fine, but the revision history doesn't indicate that it was modified to fix a race.

Reply:
Fixed. If only the other 3 functions had the same fix applied 7 years ago!